apps/api/v2/src/ee/calendars/calendars.interface.ts

```diff
@@ -3,7 +3,13 @@ import { Request } from "express";
 import { ApiResponse } from "@calcom/platform-types";
 
 export interface CalendarApp {
-  save(state: string, code: string, origin: string): Promise<{ url: string }>;
+  save(
+    state: string,
+    code: string,
+    origin: string,
+    username?: string,
+    password?: string
+  ): Promise<{ url: string } | { status: string }>;
   check(userId: number): Promise<ApiResponse>;
 }
```

Comment:
I would create a different interface for Credential based calendar apps

Reply:
Agreed!